### PR TITLE
Removed small redundant validation on gfx_s2dex_bg_copy and added ifs to avoid null textures

### PR DESCRIPTION
**Pull request by KiritoDv**

### CODE REVIEW

File: src/graphic/Fast3D/gfx_pc.cpp [hunk 1]
```diff
@@ -2243,18 +2243,16 @@ static void gfx_s2dex_bg_copy(uObjBg* bg) {
...
-            texFlags = tex->Flags;
-            rawTexMetadata.width = tex->Width;
-            rawTexMetadata.height = tex->Height;
-            rawTexMetadata.h_byte_scale = tex->HByteScale;
-            rawTexMetadata.v_pixel_scale = tex->VPixelScale;
-            rawTexMetadata.type = tex->Type;
-            rawTexMetadata.name = std::string((char*)data);
-            data = (uintptr_t) reinterpret_cast<char*>(tex->ImageData);
-        }
+    if ((bool) gfx_check_image_signature((char*)data)) {
```
Comment (briaguya0) [inline src/graphic/Fast3D/gfx_pc.cpp]:
```suggestion
    if ((bool)gfx_check_image_signature((char*)data)) {
```
make clang format happy

**Review by Kenix3 — APPROVED** [review]

**Comment by Kenix3** [comment]
Looks like we need to run clang-format on this again. After that it appears to be ready.

File: src/graphic/Fast3D/gfx_pc.cpp [hunk 2]
```diff
@@ -2243,18 +2243,16 @@ static void gfx_s2dex_bg_copy(uObjBg* bg) {
     uint32_t texFlags = 0;
     RawTexMetadata rawTexMetadata = {};
 
-    if ((data & 1) != 1) {
```
Comment (Kenix3) [inline src/graphic/Fast3D/gfx_pc.cpp:2246]: Why are you getting rid of the lowest order bit check?

Comment (KiritoDv) [reply src/graphic/Fast3D/gfx_pc.cpp:2246]: Because the gfx_check_image_signature already does this

**Pull request merged**